scripts/main.js

```diff
@@ -308,9 +314,89 @@ window.addEventListener('WebChannelMessageToContent', function (event) {
       selectNextItem(-1);
     }
   }
+
+  // collecting session metrics on open and closed message types
+  if (message.type === 'popupopen') {
+
+    sessionMetrics.set({startTime: new Date().getTime()});
+
+  } else if (message.type === 'popupclose') {
+
+    sessionMetrics.set({endTime: new Date().getTime()}); 
+    sessionMetrics.set({duration:sessionMetrics.getDuration()});
+
+    if (sessionMetrics.attributes.hasEngaged) {
+
+      var selectedGroup = $('li.selected').parent().parent();
+      var resultIndex = selectedGroup.find('.selected').index();
+
+      sessionMetrics.set({resultType: selectedGroup.attr('data-region')});
+      sessionMetrics.set({resultIndex: resultIndex});
+
+    }
+
+    console.log(sessionMetrics.attributes);
+
+  }
+
 });
 
 // Mouse events should clear arrow selection for now
 $(window).mouseenter(function () {
   $('li.selected').removeClass('selected');
 });
+
+
+$(window).mousemove(function () {
+  // get the index of the selected result before the state change
+  var lastSelectedIndex = $('.result').index($('.selected'));
+
+  // set some metrics flags
+  sessionMetrics.setEngagement();
+  sessionMetrics.setLastNavEventUsed('mouse');
+
+  //change the state
+  $('li.selected').removeClass('selected');
+  $('li:hover').addClass('selected');
+
+  // get the index of the new selected result
+  var selectedIndex = $('.result').index($('.selected'));
+
+  if (selectedIndex >= 0 && lastSelectedIndex != selectedIndex) {
+    window.dispatchEvent(new window.CustomEvent('selectionChanged', { detail: { url: $('.selected').find('.result-url').text().trim() } }));
+  }
+
+});
+
+// Model for storing session metrics.
+// Right now this just gets logged out when the popup closes
+// defaults below indicate what's currently collected
+var SessionMetrics = Backbone.Model.extend({
+  defaults: {
+    duration: undefined,
+    endTime: undefined,
+    hasEngaged: false,
+    resultIndex: undefined,
+    resultType: undefined,
+    startTime: undefined
```

Comment:
I wonder if there might be some privacy concerns around using absolute times instead of measuring the amount of time the user engaged.